tests/components/velbus/test_init.py

```diff
@@ -54,3 +55,19 @@ async def test_device_identifier_migration(
     assert device_entry.manufacturer == "Velleman"
     assert device_entry.model == "module_type_name"
     assert device_entry.sw_version == "module_sw_version"
+
+
+async def test_migrate_config_entry(hass: HomeAssistant):
+    """Test successful migration of entry data."""
+    legacy_config = {CONF_NAME: "fake_name", CONF_PORT: "1.2.3.4:5678"}
+    entry = MockConfigEntry(domain=DOMAIN, unique_id="my own id", data=legacy_config)
+    entry.version = 1
+    entry.add_to_hass(hass)
+
+    assert entry.data == legacy_config
+    assert entry.version == 1
+
+    await hass.config_entries.async_setup(entry.entry_id)
+
+    assert entry.data == legacy_config
+    assert entry.version == 1
```

Comment:
Shouldn't the version be 2 after migration?

Reply:
not yet, when we do a next upgrade of velbusaio this upgrade code will be enabled and needed.
Until then (that PR is ready) and i will submit it the moment this one is merged

Reply:
But we're setting the version to 2 in the migration. What's this test testing if the version isn't 2 after the migration?

Reply:
no not yet, currently the version is still 1....

Reply:
We're not setting up the test correctly then. Please test the whole new code. I guess you need to patch the file check.

Reply:
I fixed the tests. We still need to add some patching around the file handling.